### ISSUE TITLE
Free Subscription & Classes Diagram

### ISSUE THREAD

**Issue by belle101** [Simbul/baker#874]
Hi,

I want to ask about free subscription
what does it actually do? first i thought that it will give you info if there's a new issue 
but when i subscribe and add a new issue to shelf
there's nothing happen,

thanks

**Comment by nicholasmartin** [comment]
I'm totally NOT an expert on this, but I think you need to setup the new issue in Itunes Connect under Manage Newsstand first, before the app will be sent a notification of a new issue. This happens only once a day.

**Comment by Simbul** [comment]
On its own, free subscription does nothing - other than recording the fact that a user has subscribed.
When you say "it will give you info if there's a new issue", I assume you're referring to Push Notifications. Unfortunately, push notifications require an external server and you will have to send them yourself (i.e. Apple is not doing it automatically). Have a look at the [Push notifications tutorial](https://github.com/Simbul/baker/wiki/Push-notifications-tutorial).

To summarise: when a user subscribes, Baker will know they did it, and tell your server. It is your server's responsibility to send a push notification to subscribed users when a new issue comes out.

**Comment by belle101** [comment]
So, can i just disable the button?

**Comment by Simbul** [comment]
That's a thorny issue. Apple store guidelines mandate that if you have a Newsstand app, then at least a free subscription should be offered.

I think I remember some Baker apps being approved in the past even though they had no push notifications, so you may want to leave the subscription button there and try your luck.

From the user's point of view, if your issues are all free, being subscribed or not won't make a difference (either positive or negative).

**Comment by belle101** [comment]
Ok alright then i'll just leave it there,
also i'm curious is there any explanation of what is all the .m and .h files are for?
i'm currently using baker for my assignment at my college so i need to analyze it :s

anyway
thanks for your answer(s) really appreciate it :)

**Comment by folletto** [comment]
What do you mean "all the .m and .h files are for"? :)

**Comment by belle101** [comment]
Haha i read my reply again and it IS confusing, sorry,
i mean the documentation, because i see things like reachability, bytereceived etc in the script but i didn't know how to use it because there's no comment of what it is for 

I'm sorry if you still confused of what i mean, i'm not really familiar with objective-c so trying to explain about the code for my assignment is a bit hard 

What i'm trying to say is i have to explain about bakerissue.m, issuemanager.m, and all the m files for my assignment (like creating flow charts and all)

Sorry if this is confusing :(

**Comment by folletto** [comment]
This is a high-level view of the main classes relationships. It doesn't map every single call (for example `PurchaseManager` is a singleton so technically doesn't belong to anyone) but shows the main hierarchy to get a first level understanding of the framework. :)

![baker-overview-classes-diagram](https://f.cloud.github.com/assets/4389/498825/c7b84bf0-bc20-11e2-8d9d-391d6d6ee881.png)

**Comment by belle101** [comment]
Thank you soooooo much!!! 

:)
Sent from my BlackBerry®
powered by Sinyal Kuat INDOSAT

-----Original Message-----

**Comment by belle101** [comment]
I'm sorry i have to open this again, 
i want to ask about the subscription mechanism,
when user subscribe (even free subscribe) they need to put their apple id and password,
and after that there are 2 option
- letting the developer know their info
- not letting the developer know their info
  how's the mechanism? because i don't have any server so where do their info go? 
  thankssssss

**Comment by folletto** [comment]
If you don't have any server, the info doesn't go anywhere. :)
Otherwise, build yours and the info will go there. Check the wiki for the Baker API. ;)
